arch/loongarch/crc32_la.c

```diff
@@ -0,0 +1,51 @@
+/* crc32_la.c - LoongArch version of crc32
+ * Copyright (C) 2025 Vladislav Shchapov <vladislav@shchapov.ru>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#if defined(LOONGARCH_CRC)
+#include "zbuild.h"
+#include "zmemory.h"
+#include "zutil.h"
+#include "crc32.h"
+#include <stdint.h>
+
+#include <larchintrin.h>
+
+Z_INTERNAL uint32_t crc32_loongarch64(uint32_t crc, const uint8_t *buf,
+                                      size_t len) {
+    crc = (~crc) & 0xffffffff;
+
+    while (len >= 8) {
+        crc = (uint32_t)__crc_w_d_w((long int)zng_memread_8(buf), (int)crc);
```

Comment:
Does Loongarch have strict alignment requirements for loads and stores? If it does, you might want to do what the ARMv8 implementation does to avoid loading words one byte at a time. If not, it might be worth setting `OPTIMAL_CMP` in `zbuild.h` to a suitable value to gain some potential performance benefits on machines without LSX/LASX.

Reply:
Unaligned access is allowed by the implementation.

It sounds reasonable. I changed the crc32 implementation to align the buffer.

I believe that choosing the `OPTIMAL_CMP` value for the platform is beyond the scope of this PR.